catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/CswQueryResponseTransformer.java

```diff
@@ -255,54 +262,65 @@ private String convert(CswRecordCollection cswRecordCollection, List<Result> res
+            throws CatalogTransformerException {
 
         CompletionService<BinaryContent> completionService = new ExecutorCompletionService<>(
                 queryExecutor);
 
-        try {
-            final MetacardTransformer transformer =
-                    metacardTransformerManager.getTransformerBySchema(recordSchema);
-            if (transformer == null) {
-                throw new CatalogTransformerException(
-                        "Cannot find transformer for schema: " + recordSchema);
-            }
+        final MetacardTransformer transformer = metacardTransformerManager.getTransformerBySchema(
+                recordSchema);
+        if (transformer == null) {
+            throw new CatalogTransformerException(
+                    "Cannot find transformer for schema: " + recordSchema);
+        }
 
-            Map<Future<BinaryContent>, Result> futures = new HashMap<>(results.size());
-            for (Result result : results) {
-                final Metacard mc = result.getMetacard();
+        Map<Future<BinaryContent>, Result> futures = new HashMap<>(results.size());
+        for (Result result : results) {
+            final Metacard mc = result.getMetacard();
 
-                // the "current" thread will run submitted task when queueSize exceeded; effectively
-                // blocking enqueue of more tasks.
-                futures.put(completionService.submit(() -> {
-                    BinaryContent content = transformer.transform(mc, arguments);
-                    return content;
-                }), result);
-            }
+            // the "current" thread will run submitted task when queueSize exceeded; effectively
+            // blocking enqueue of more tasks.
+            futures.put(completionService.submit(() -> {
+                BinaryContent content = transformer.transform(mc, arguments);
+                return content;
+            }), result);
+        }
 
-            // TODO - really? I can't use a list of some sort?
-            InputStream[] contents = new InputStream[results.size()];
+        InputStream[] contents = new InputStream[results.size()];
 
-            while (!futures.isEmpty()) {
+        while (!futures.isEmpty()) {
+            try {
                 Future<BinaryContent> completedFuture = completionService.take();
                 int index = results.indexOf(futures.get(completedFuture));
-                contents[index] = completedFuture.get()
-                        .getInputStream();
-                futures.remove(completedFuture);
-            }
-
-            CharArrayWriter accum = new CharArrayWriter(ACCUM_INITIAL_SIZE);
-            for (InputStream is : contents) {
-                IOUtils.copy(is, accum);
+                try {
+                    contents[index] = completedFuture.get()
+                            .getInputStream();
+                } catch (ExecutionException e) {
```

Comment:
Since the execution here is invoking transform on a metacard transformer, the only declared exception that can be thrown is `CatalogTransformerException`.  However, it's not possible to catch this exception directly because of the use of `Future`s